.github/workflows/build-ci.yml

```diff
@@ -16,56 +16,102 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - name-prefix: "C++ without MPI"
+          - name-prefix: "Python 3.6 without MPI"
             enable-mpi: false
-            configure-options: "--without-python"
-            python-version: "none"
-          - name-prefix: "C++ with MPI"
-            enable-mpi: true
-            configure-options: "--without-python --with-mpi"
-            python-version: "none"
-          - name-prefix: "Python 3.6"
-            enable-mpi: false
-            configure-options: "--with-python --enable-maintainer-mode"
+            configure-options: "--without-mpi"
             python-version: "3.6"
-          - name-prefix: "Python 3.9"
+          - name-prefix: "Python 3.9 without MPI"
             enable-mpi: false
-            configure-options: "--with-python --enable-maintainer-mode"
+            configure-options: "--without-mpi"
+            python-version: "3.9"
+          - name-prefix: "Python 3.6 with MPI"
+            enable-mpi: true
+            configure-options: "--with-mpi"
+            python-version: "3.6"
+          - name-prefix: "Python 3.9 with MPI"
+            enable-mpi: true
+            configure-options: "--with-mpi"
             python-version: "3.9"
```

Comment:
If this is a true Cartesian product of MPI x Python version, you don't have to rely on `include` to build the four cases manually: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix